File: sdks/dotnet/src/Mobiscroll.Connect/Exceptions/NetworkException.cs
using System;

namespace Mobiscroll.Connect.Exceptions;

public sealed class NetworkException : MobiscrollConnectException
{
    public NetworkException(string message) : base(message, "NETWORK_ERROR") { }

    public NetworkException(string message, Exception innerException)
        : base(message, "NETWORK_ERROR", innerException) { }
}

File: sdks/java/connect-sdk/src/main/java/com/mobiscroll/connect/exceptions/NetworkException.java
package com.mobiscroll.connect.exceptions;

/** Thrown on transport-level failures (timeouts, DNS, connection reset, etc.). */
public class NetworkException extends MobiscrollConnectException {

    private static final long serialVersionUID = 1L;

    public NetworkException(String message, Throwable cause) {
        super(message, cause);
    }
}
